src/overlays/ovl_i2/fox_me.c

```diff
@@ -1,97 +1,2527 @@
+    actor->obj.rot.y += 7.0f;
+    actor->obj.rot.x += 3.3f;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018DA6C.s")
+    if (actor->unk_0D0 != 0) {
+        func_80066254(actor);
+        func_8007A6F0(&actor->obj.pos, 0x2903A008);
+        Object_Kill(&actor->obj, &actor->sfxPos);
+        func_8007D0E0(actor->obj.pos.x - actor->vel.x, actor->obj.pos.y, actor->obj.pos.z - actor->vel.z, 8.0f);
+        func_8007BFFC(actor->obj.pos.x - actor->vel.x, actor->obj.pos.y + 30.0f, actor->obj.pos.z - actor->vel.z, 0.0f,
+                      0.0f, 0.0f, 4.0f, 10);
+    }
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018DAEC.s")
+    if (!(gGameFrameCount & 7)) {
+        Math_Vec3fFromAngles(&vec, actor->obj.rot.x, actor->obj.rot.y, 100.0f);
+        func_8007F04C(0x161, actor->obj.pos.x + (vec.x * 3.0f), actor->obj.pos.y + (vec.y * 3.0f),
+                      actor->obj.pos.z + (vec.z * 3.0f), actor->obj.rot.x, actor->obj.rot.y, actor->obj.rot.z, 0.0f,
+                      0.0f, 0.0f, vec.x, vec.y, vec.z, 1.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018DB6C.s")
+void func_i2_80187B08(Actor* actor) {
+    actor->obj.rot.y += 7.0f;
+    actor->obj.rot.x += 3.3f;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018DBEC.s")
+    actor->unk_046 -= 11;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018DCE4.s")
+    if (actor->unk_046 < 0) {
+        actor->unk_046 = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018DE14.s")
+    actor->unk_048 += 4;
+    if (actor->unk_048 > 995) {
+        actor->unk_048 = 995;
+    }
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018DF08.s")
+    if (!(actor->timer_0BC & 3)) {
+        func_8007D0E0(actor->obj.pos.x, actor->obj.pos.y, actor->obj.pos.z, 6.0f);
+    }
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018E084.s")
+    if ((actor->unk_0D0 != 0) || (actor->timer_0BC == 0)) {
+        func_8007A6F0(&actor->obj.pos, 0x2903A008);
+        Object_Kill(&actor->obj, &actor->sfxPos);
+        func_8007D0E0(actor->obj.pos.x - actor->vel.x, actor->obj.pos.y, actor->obj.pos.z - actor->vel.z, 8.0f);
+        func_8007BFFC(actor->obj.pos.x - actor->vel.x, actor->obj.pos.y + 30.0f, actor->obj.pos.z - actor->vel.z, 0.0f,
+                      0.0f, 0.0f, 4.0f, 10);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018ED9C.s")
+void func_i2_80187C68(Actor* actor, f32 x, f32 y, f32 z, f32 arg4, f32 xRot, f32 yRot, s32 arg7, s32 arg8, s32 id) {
+    Actor_Initialize(actor);
+    actor->obj.status = 1;
+    actor->obj.id = (u16) id;
+
+    actor->obj.pos.x = x;
+    actor->obj.pos.y = y;
+    actor->obj.pos.z = z;
+
+    actor->obj.rot.x = xRot;
+    actor->obj.rot.y = yRot;
+
+    actor->timer_0BC = arg7;
```

Comment:
For arguments like these, I like to name them after the field they're being assigned to. So timerBC or unkB4. Gives a little more info than just argN.